### DISCUSSION TITLE
Error while opening the remote_controlboard of the icubSim head

### DISCUSSION THREAD

**Discussion by Sohaib448506** [robotology/community#463] [Q&A] [answered]
Hi World, I installed ICUB and yarp from the platform of robotology. I run the code and it's work. But one of mine friend did the same process as given in Robotology and run the code. But he is getting error . Kindly have a look and guide me how to solve it.

**Code:**
```matlab
function gazeShift_iCub()  
[objcoorPort,objs,portLeft,portRight,headDriver,headPos,headEncs]=yarpInit();
%%  Initialization
stepSize=1;
allowedPans=[-20:stepSize:+20]; 
allowedTilts=[-12:stepSize:+12]; 
headcentPans=[-20:0.1:+20]; 
headcentTilts=[-12:0.1:+12]; 
for k=1:length(allowedPans)
            for l=1:length(allowedTilts)
                panVal=allowedPans(randi(length(allowedPans)));
                tiltVal=allowedTilts(randi(length(allowedTilts)));
                headPos.positionMove(4,panVal); % Joint 4 both Eyes pan/version
                headPos.positionMove(3,tiltVal); % Joint 3 both Eyes tilt
                pause(60);
                headPos.positionMove(4,0); % Joint 4 both Eyes pan/version
                headPos.positionMove(3,0); % Joint 3 both Eyes tilt
                pause(3); 
            end
end
%% Functions
function [objcoorPort,objs,portLeft,portRight,headDriver,headPos,headEncs]=yarpInit()
yarp.matlab.LoadYarp;
import yarp.Port;
import yarp.Bottle;
import yarp.BufferedPortImageRgb
import yarp.BufferedPortBottle;
portLeft=BufferedPortImageRgb;
portRight=BufferedPortImageRgb;
portLeft.open('/matlab/LeftEye');
yarp.Network.connect('/icubSim/cam/left/fovea','/matlab/LeftEye');
portRight.open('/matlab/RightEye');
yarp.Network.connect('/icubSim/cam/right/fovea','/matlab/RightEye');
yarp.Network.init();
%  prepare a property object
headOptions = yarp.Property();
headOptions.put('device','remote_controlboard');
headOptions.put('local','/client/head');
headOptions.put('remote','/icubSim/head');
%  create remote driver
headDriver = yarp.PolyDriver(headOptions);
headPos = headDriver.viewIPositionControl();
headEncs = headDriver.viewIEncoders();
joints=headPos.getAxes();    % get the number of controlled axes in joints
% %print 'Controlling', 'joints'
encoders=yarp.Vector(joints);
headEncs.getEncoders(encoders.data());
objcoorPort=Port;
objcoorPort.open('/matlab/objcoordTx');
%objs=Bottle; 
objs=Bottle;
```

**Following the result of same code which is working at my computer.**

**Error:**

```text
coordSaccadeAndVergenceLearn_iCub
Yarp library loaded and initialized
[INFO] |yarp.os.Port| Port /matlab/LeftEye active at
tcp://192.168.10.10:10054/
[INFO] |yarp.os.Port| Port /matlab/RightEye active at
tcp://192.168.10.10:10055/
[ERROR] |yarp.os.YarpPluginSettings| Cannot find "remote_controlboard"
plugin (not built in, and no .ini file found for it)Check that
YARP_DATA_DIRS leads to at least one directory with
plugins/remote_controlboard.ini or
share/yarp/plugins/remote_controlboard.ini in it
[ERROR] |yarp.dev.PolyDriver| Could not find device <remote_controlboard>
Dot indexing is not supported for variables of this type.

Error in coordSaccadeAndVergenceLearn_iCub>yarpInit (line 319)
joints=headPos.getAxes();    % get the number of controlled axes in joints

Error in coordSaccadeAndVergenceLearn_iCub (line 6)
[objcoorPort,objs,portLeft,portRight,headDriver,headPos,headEncs]=yarpInit();
```

Thanks...............................

**Comment by pattacini** [comment, marked as answer]
Very likely, if you run the command below
```sh
$ yarpdev --list | grep remote_controlboard
```
you won't find any reference, meaning that YARP is unable to find the necessary information to instantiate the device driver network wrapper.

A possible solution is suggested in the output you reported:
```text
Check that YARP_DATA_DIRS leads to at least one directory with
plugins/remote_controlboard.ini or
share/yarp/plugins/remote_controlboard.ini in it
```

If you install the software using the [`robotology-superbuild`](https://github.com/robotology/robotology-superbuild), this will be taken care of. By contrast, you ought to set up the `YARP_DATA_DIRS` env variable pointing to a proper path under which that particular ini file is stored.

**Reply by traversaro** [reply]
> If you install the software using the [`robotology-superbuild`](https://github.com/robotology/robotology-superbuild), this will be taken care of. By contrast, you ought to set up the `YARP_DATA_DIRS` env variable pointing to a proper path under which that particular ini file is stored.

Yes, that should be the case, in particular if you correctly sourced the `setup.sh` or equivalent environment script, as described in https://github.com/robotology/robotology-superbuild#configure-your-environment (assuming that you are using Ubuntu). Given that you are using MATLAB, you should make sure that you are opening MATLAB in a way that it gets the environment variables set by the `setup.sh` script, for example my launching it from a terminal in which the `setup.sh` script was sourced.

To summarize, I think that if you could provide the following info it would be easier for us to help you:
* Which operating system you are using? 
* How did you configured your environment? 
* How are you launching MATLAB?

**Reply by Sohaib448506** [reply]
@traversaro and @pattacini ..Thanks Issues resolved by opening the MATLAB from terminal.
